puppet/chassis.rb

```diff
@@ -282,4 +292,127 @@ def self.make_relative(base, paths)
 			pathname.relative_path_from(base).to_s
 		end
 	end
+
+	def self.update_core(config)
+		if config["auto_update"]["core"] == true
+			puts "\e[32mChecking for Chassis core updates...\e[0m"
+			updates = self.updates_check(['core'], @@dir)
+			if updates.empty?
+				puts "\033[A\e[K"
+			else
+				self.prompt_for_updates(updates, @@extension_dir, 'core')
+			end
+		end
+	end
+
+	def self.update_submodules(config)
+		if config["auto_update"]["submodules"] == true
+			puts "\033[A\e[32mChecking for Chassis submodule updates...\e[0m"
+			submodules = ['apt','mysql','stdlib','wp']
+			directory = File.join(@@dir, 'puppet/modules')
+			updates = self.submodule_update_check(submodules, directory)
+		end
+	end
+
+	def self.update_extensions(config)
+		if config["auto_update"]["extensions"] == true
+			puts "\033[A\e[32mChecking for Chassis extension updates...\e[0m"
+			extensions = Dir.glob(@@extension_dir + '/*').map { |directory| File.basename( directory ) }
+			updates = self.updates_check(extensions, @@extension_dir)
+			if updates.empty?
+				puts "\033[A\r\e[K"
+			else
+				self.prompt_for_updates(updates, @@extension_dir, 'extensions')
+			end
+		end
+	end
+
+	def self.update_global_extensions(config)
+		if config["auto_update"]["global_extensions"] == true
+			puts "\033[A\e[32mChecking for Chassis global extension updates...\e[0m"
+			global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+			global_extensions = Dir.glob(global_ext_path + '/*').map { |directory| File.basename( directory ) }
+			updates = self.updates_check(global_extensions, global_ext_path)
+			if updates.empty?
+				puts "\033[A\r\e[K"
+			else
+				self.prompt_for_updates(updates, global_ext_path, 'extensions')
+			end
+		end
+	end
+
+	def self.updates_check(folders, directory)
+		updates = Array.new
+		folders.each do |folder|
+			next if folder == 'example'
+			if 'core' == folder
+				folder = directory
+				Dir.chdir(directory)
+			elsif
+				Dir.chdir(directory + '/' + folder )
+			end
+			Open3.capture3("git remote -v update")
+			stdout, _, _ = Open3.capture3("git status --porcelain=2 --branch")
+			if stdout =~ /^# branch.ab \+\d+ -([1-9]\d*)$/
```

Comment:
This check should be simplified down as right now only changes where lines of code have been deleted will register as an update. This can be:
```suggestion
			if stdout !~ /^# branch.ab \+0 -0/
```

Reply:
Good catch Sir! 💖